Dev 3.2

GigaZiv opened the pull request:
Переработал большинство модулей так как не нравится их реализация. Проблема в том, что не ясно какие отставлять "заделы" на будущее, так как задачи не ясны.

GigaZiv committed:
Dev 3.2

Переработал большинство модулей так как не нравится их реализация. Проблема в том, что не ясно какие отставлять "заделы" на будущее, так как задачи не ясны.

IliaAnK commented on app/src/main/java/rs/example/playlistmaker/SearchAcitivty.kt:
```diff
@@ -1,105 +1,144 @@
+
+            etSearch.apply {
+                doAfterTextChanged { s ->
+                    iwClear.visibility = clearVisibility(s)
```
Рекомендация:
Можно сделать чуть проще: `iwClear.isVisible = !s.isNullOrEmpty()`
А метод `clearButtonVisibility`, соответственно, удалить

GigaZiv replied:
Согласен

IliaAnK commented on app/src/main/res/layout/activity_search.xml:
```diff
@@ -14,7 +14,7 @@
         android:layout_height="?attr/actionBarSize"
         android:gravity="center_vertical"
-        android:text="@string/search" />
+        android:text="@string/с_search" />
```
Рекомендация:
В качестве заголовка экрана лучше использовать `MaterialToolbar`. Это стандартный компонент. В него можно передать текст заголовка с помощью атрибута `title`, иконку(стрелочка влево) с помощью атрибута `app:navigationIcon` и обработать нажатие с помощью `setNavigationOnClickListener {}`. Характеристики текста можно задать через стиль и атрибута `titleTextAppearance`.
Для остальных экранов аналогично.

GigaZiv replied:
А я об этом с первого урока говорил, но на сколько я помню нельзя было использовать MaterialToolbar

IliaAnK replied:
Не в курсе таких запретов. Думаю, что можно и нужно.

IliaAnK commented:
Экран настройки:

Из общего: думаю тут в качестве корневого `layout` лучше использовать `LinearLayout`, чтобы все дочернии `View` не приходилось "оборачивать" в `TableRow`. Внутри `TableRow` представлен как `LinearLayout` и тут появляются возможные "просадки" по отрисовки UI, поскольку в это время каждая `ViewGroup` создаёт собственную иерархию, измеряет себя и встраивается в родительский контейнер.

**Итого:**
1. Излишняя вложенность негативно влияет на скорость отрисовки фрейма.
2. Она может вызывает трудности с понимание того как свёрстан экран;
3. Без неё проще формировать и применять стили напрямую к `View`

В качестве заголовка экрана лучше использовать `MaterialToolbar `. Это стандартный компонент. В него можно передать текст заголовка, иконку(стрелочка влево) с помощью атрибута `app:navigationIcon` и обработать нажатие с помощью `setNavigationOnClickListener {}`. Характеристики текста можно задать через стиль и атрибута `titleTextAppearance`.

Далее, если решишь перейти с `TableLayout` на `LinearLayout`, то:
```
<TableRow> 
    <TextView/>
    <Switch/> 
</TableRow>
```
Можно представить как:
```
<com.google.android.material.switchmaterial.SwitchMaterial
    android:layout_width="match_parent"
    android:layout_height="wrap_content"
    android:gravity="center_vertical"
    android:paddingHorizontal="16dp"
    android:paddingVertical="20dp"
    android:text="@string/c_theme_dark"
    app:thumbTint="..." // Нужен для стилизации свичера
    app:trackTint="..." // Нужен для стилизации свичера
    android:textSize="16sp" />
```

`TextView` будет просто лежать в корне `LinearLayout`'а

Общие характеристика вынести само собой можно вынести в стили.

Итого иерархия всего `layout`-а будет вот такой:
```
<LinearLayout>
    <MaterialToolbar/>
    <SwitchMaterial/>
    <MaterialTextView/>
    <MaterialTextView/>
    <MaterialTextView/>
<LinearLayout/>
```

GigaZiv committed:
Dev 3.3
1. Переработал (два) Activity's добавлены MaterialToolbar's. Соответственно добавлены стили для app:navigationIcon светлой и темной темы, обработаны события кнопки Назад;
2. ...На экране «Поиск» в поле ввода цвет курсора... - Добавлены соответствующие ресурсы. Если конечно действительно надо чтобы цвет самого курсора был - #3772E7;
3. ...На экране «Настройки» стиль свичера... привел в соответствия цвета;
4. В коде добавлена поддержка setSupportActionBar;
5. Добавлена функция определения темы приложения и ОС системы - getModeNightAppUI;
6. Скорректирован код, согласно Вашей рекомендации на github.
7. Добавил в Activity Настроек
<item name="android:background">?attr/selectableItemBackground</item>
для Ripple эффекта нажатия на пункты

С уважением Жуков Игорь
